# Exploratory Data Analysis
Tujuan: Menganalisis pola dan perilaku pada data untuk menjawab beberapa pertanyaan.

## 1. Data Loading
Pertama, kita perlu mengambil dan melihat data. Saat ini, kita sudah bisa menggunakan data yang telah diproses pada notebook sebelumnya.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
results = pd.read_csv("../data/interim/results_clean.csv")

### Preview Dataset

In [10]:
results.head()

,date,home_team,away_team,home_score,away_score,tournament,city,country,neutral
0,1872-11-30,Scotland,England,0.0,0.0,Friendly,Glasgow,Scotland,False
1,1873-03-08,England,Scotland,4.0,2.0,Friendly,London,England,False
2,1874-03-07,Scotland,England,2.0,1.0,Friendly,Glasgow,Scotland,False
3,1875-03-06,England,Scotland,2.0,2.0,Friendly,London,England,False
4,1876-03-04,Scotland,England,3.0,0.0,Friendly,Glasgow,Scotland,False


## 2. Dataset Overview

Kita akan mulai dengan mengenali dataset kita.

### Pertanyaan 1:
**Seberapa besar dataset yang kita analisis?**

In [3]:
print(f"Jumlah baris : {results.shape[0]:,}")
print(f"Jumlah kolom : {results.shape[1]}")

Jumlah baris : 49,433
Jumlah kolom : 9


### Insight 1:
Dataset ini memiliki lebih dari 49 ribu hasil pertandingan internasional dengan berbagai atribut penting seperti tanggal pertandingan, tim, skor, turnamen, dan lokasi. Jumlah observasi yang besar memberikan landasan yang kuat untuk pemodelan Machine Learning. Langkah berikutnya adalah memahami distribusi karakteristik data, terutama keseimbangan kelas hasil pertandingan.

### Pertanyaan 2: 
**Bagaimana rentang waktu dataset ini?**

In [ ]:
print(f"Tanggal pertama: {results['date'].min()}")
print(f"Tanggal terakhir: {results['date'].max()}")

Tanggal pertama: 1872-11-30
Tanggal terakhir: 2026-06-18


### Insight 2:
Rentang waktu 154 tahun (1872–2026) tidak hanya menunjukkan kekayaan sejarah, tetapi juga memberikan tantangan analitis. Aturan, taktik, dan intensitas sepak bola telah berevolusi secara drastis; pola historis di awal abad ke-20 mungkin tidak lagi relevan dengan dinamika sepak bola modern. Untuk pemodelan, kita dapat mempertimbangkan untuk membatasi rentang waktu pelatihan atau melakukan analisis per era apabila ditemukan perubahan pola yang signifikan.

### Pertanyaan 3: 
**Berapa banyak tim unik?**

In [6]:
teams = pd.concat([
    results["home_team"],
    results["away_team"]
])

print(f"Banyak tim berbeda: {teams.nunique()}")

Banyak tim berbeda: 336


### Insight 3:
Dataset memuat 336 entitas tim yang berbeda, jumlah yang melebihi keanggotaan FIFA saat ini. Hal ini mencerminkan perubahan geopolitik sepanjang sejarah, seperti pemecahan negara maupun pergantian nama negara. Oleh karena itu, proses standardisasi nama tim akan menjadi langkah penting pada tahap feature engineering.

### Pertanyaan 4: 
**Berapa banyak jenis turnamen?**

In [8]:
print(results["tournament"].nunique())

200


In [9]:
results["tournament"].value_counts().head(10)

tournament
Friendly                                18388
FIFA World Cup qualification             8771
UEFA Euro qualification                  2824
African Cup of Nations qualification     2327
FIFA World Cup                            992
Copa América                              869
African Cup of Nations                    845
AFC Asian Cup qualification               829
UEFA Nations League                       658
CECAFA Cup                                620
Name: count, dtype: int64

### Insight 4
Dataset ini merangkum 200 variasi turnamen, yang didominasi oleh Friendly Match dan babak kualifikasi. Tingkat kompetisi sangat memengaruhi hasil laga, tim besar cenderung merotasi pemain dan tampil kurang maksimal di laga persahabatan dibandingkan saat bermain di Piala Dunia. Salah satu pendekatan yang layak dieksplorasi adalah mengelompokkan turnamen berdasarkan tingkat prestise agar model lebih mudah menangkap konteks pertandingan.

### Implikasi untuk Tahap Berikutnya

Secara keseluruhan, eksplorasi awal ini memberikan kita gambaran yang kuat mengenai kondisi data dan apa yang perlu dilakukan sebelum data ini siap masuk ke tahap permodelan. Berikut adalah ringkasan dari temuan kita:

| Pertanyaan | Temuan Faktual (Nominal) | Insight & Tindakan Lanjutan (*Action Plan*) |
|---|---|---|
| **Seberapa besar dataset yang dianalisis?** | **> 49.000 baris** dengan beberapa kolom atribut. | Volume data sangat mumpuni untuk prediksi ML. Bisa cek distribusi kelas (*class imbalance*) dari hasil akhir laga. |
| **Bagaimana rentang waktu dataset ini?** | **154 tahun** (1872 - 2026). | Pola lama mungkin *outdated*. Butuh pembagian data latih-uji secara kronologis (*time-based split*) dan clustering era. |
| **Berapa banyak tim unik?** | **336 tim** berbeda. | Melebihi batas keanggotaan FIFA, butuh pemetaan sejarah geopolitik/perubahan nama negara. |
| **Berapa banyak jenis turnamen?** | **200 variasi** turnamen. | Gengsi kompetisi memengaruhi hasil laga. Pendekatan yang layak dieksplorasi adalah mengelompokkan turnamen berdasarkan tingkat prestise agar model lebih mudah menangkap konteks pertandingan. |